Toggle switch

Absotively opened the pull request:
Adds switch button styles, converts APC power toggle to switch button, replaces `CheckButton` with `Button` in three places.

Requires https://github.com/space-wizards/RobustToolbox/pull/6099

## About the PR
This adds styles for switch buttons and changes the APC power toggle to use a switch button.

I am leaving other power buttons/toggles for a later PR on the theory that how the button should look and where it should be used are mostly separate questions.

## Why / Balance
The current toggle button for APC power only displays the status through the button colour. Also, I think "toggle" is not a very helpful button label. Using a switch-style button with state labels will hopefully be clearer.

Most or all of the gas manipulation devices also have unclear power toggling buttons, though they are different unclear power toggling buttons. This new control will hopefully be useful for them too, but for the sake of keeping this PR manageable they will be changed later rather than here.

There may be other off/on buttons that would be clearer if they were switch-style buttons, but many of them are two-button off/on groups, which are less unclear and so of lower priority to change.

## Technical details
Textures and style rules for `SwitchButton` have been added to Resources/Textures/Interface/Nano and StyleNano.cs.

The APC menu has been changed to use a `SwitchButton`, and the string for the old toggle button's label has been removed.

Three instances of `CheckButton` in xaml files were changed to `Button`. This is not expected to cause any visible change, since `CheckButton` is essentially a now-obsolete synonym for `Button`. See https://github.com/space-wizards/RobustToolbox/pull/6099 for more details.

## Media
The new control in the APC UI:
<img width="336" height="212" alt="image" src="https://github.com/user-attachments/assets/44d11189-5e0a-4d79-af0b-22095c854dc9" />
<img width="345" height="221" alt="image" src="https://github.com/user-attachments/assets/5e9f3ed0-1e84-4f5e-8a09-351b605844ab" />

How the new control looks with the styles added here when it has a main label:
<img width="197" height="71" alt="image" src="https://github.com/user-attachments/assets/34bd6e2b-1c10-4407-84c2-e89400a912d9" />
(No instances of the new control with a main label are being added here, but when I replace gas manipulation device power switches in a later PR, I intend to convert the "Status:" labels to `SwitchButton` main labels.)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
`CheckButton` is now obsolete. Any uses of `CheckButton` should be replaced with `Button`.

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- tweak: APC power toggle now uses switch-style button

Absotively commented:
Upon further reflection, this could probably use a second style that is less attention-grabbing.

One of the removed `CheckButton` uses was the "Show Track Names" button on the midi channels dialog. Using an actual switch button for it might be an improvement to that dialog, but I think the bright green power indicator added here would be a bit much.
<img width="240" height="335" alt="image" src="https://github.com/user-attachments/assets/9f67660a-e2ca-40fe-888f-6ccb33a32bc3" />

Probably there are other non-power things that would work well as switch buttons and benefit from a plainer style. I'll try to add a second style that harmonizes better with the standard control styles tonight.

I do think the bright indicator is probably good for power switches, which is why my plan is to add a second style rather than toning down this one.

VerinSenpai commented:
I like the switch. I feel like the internal box could benefit from thicker lines.

Personally I feel like this looks a bit jank on the APC panel. I think itd look better to have the toggle remain a toggle button and when we eventually get networked device control on apcs have this be the breakers

Absotively commented:
Textures take two: I have made these smaller, added a default style that uses the checkbox colours, and adjusted the power switch style.
<img width="213" height="127" alt="image" src="https://github.com/user-attachments/assets/f891f55e-bb0f-415f-93d1-cf12140f0bb5" />

The smaller icon fits a little better in the APC panel.
<img width="319" height="202" alt="image" src="https://github.com/user-attachments/assets/e82601db-67f8-41a0-8470-cd1a6773bfd1" />

I changed the midi channels menu's "Show Track Names" button to a switch to provide an example of the new default style.
<img width="245" height="119" alt="image" src="https://github.com/user-attachments/assets/8109cab9-412e-425c-bf46-c09f82a717e7" />

The red in the off power switch is a bit darker now, which increases the difference in brightness between the two states of the power version in the tritanopia simulator a bit. The lighter background in the on state also helps the states look more distinct.
<img width="873" height="155" alt="image" src="https://github.com/user-attachments/assets/3ead8ba2-9ac2-4c11-9f99-6ad95d54c670" />

Absotively commented:
Yet another round of reflection has yielded the following thoughts:
* These need to sit a little further down in relation to the label text. Looking at checkboxes, I think the way to do this is probably just to make the images taller but have the top part be transparent.
* Maybe they should be even shorter? Will need to make an even shorter version to compare.

Absotively commented:
Also worth noting that I will need to update this if https://github.com/space-wizards/space-station-14/pull/29903 is merged before it, which is looking likely.

sowelipililimute commented:
I'm still not too keen on the textures, I might give them an attempt

Absotively commented:
> i'm thinking something more in line with our sliders & other UI controls in size & styling, thoughts?

I like these a lot! I didn't make the thumb full height because I wanted it to look very distinct from slider controls, but I think the border color difference you've used accomplishes that very well.

I have some reservations about increasing the size, because in my opinion making the icons shorter made them look much less out of place between the two text labels. But it's certainly worth experimenting with.

VerinSenpai commented:
I think that last iteration is your best work yet

Absotively commented:
Decided to see how @sowelipililimute's design looks in context, and I think I may have been wrong about taller images not working as well.
<img width="280" height="321" alt="image" src="https://github.com/user-attachments/assets/9baf3db6-bca0-42bd-8ceb-0724f7ce4970" />

I think this is probably fine for both power and non-power uses, though; the I/O are a bit power-specific but not

Absotively commented:
Decided to see how @sowelipililimute's design looks in context, and I think I may have been wrong about taller images not working as well.
<img width="280" height="321" alt="image" src="https://github.com/user-attachments/assets/9baf3db6-bca0-42bd-8ceb-0724f7ce4970" />

I think this is probably fine for both power and non-power uses; the I/O are a bit power-specific but I think they'll work even for switches which have different state labels.

I haven't committed my implementation of @sowelipililimute's design yet, but I probably will after making disabled versions.

Absotively commented:
With disabled versions:
<img width="261" height="149" alt="image" src="https://github.com/user-attachments/assets/b21fc937-131d-42b7-bfb1-0d568b801340" />

(still haven't actually committed this, but soon!)

beck-thompson commented:
Those looks so cool :heart_eyes_cat:

sowelipililimute reviewed (approved):
this is looking fine from the content side now

Absotively commented:
> Those look damn good. Honest opinion is that the On Off should be removed so just the switch does the talking.

I think the On/Off state labels are likely to improve clarity more often than not, so I'm inclined to leave them there by default. They can now be removed from specific switches by setting the text for both to empty strings.

VerinSenpai commented:
Honestly not sure how much more clear an on off switch can get. I do believe there should be a colorblindness setting but a switch is a switch, it only has two settings.

VerinSenpai commented:
Oh... Message is gone. Now I'm talking to myself

juliangiebel reviewed (changes requested):
Sadly this PR got hit by the StyleNano removal 😔
You have to move the styles you added to StyleNano to sheetlets but once you did that this PR is good to get merged 👍

Absotively commented:
Current draft of new colours, using the Nanotrasen palette, and also the syndicate palette from [41041].
<img width="489" height="196" alt="switch_recolor_test" src="https://github.com/user-attachments/assets/0418904d-0119-48a4-8811-da97482b876f" />

I don't think these colours work as well for this, but I suppose they will fit better with the rest of the UI.

Modifying the control so all the colours can be modified at runtime, instead of by modifying icon images, may take a little longer.

Absotively commented:
Right, I think this is actually ready for re-review now!
